src/libs/UnreadIndicatorUpdater/index.js

```diff
@@ -1,38 +1,29 @@
-import {InteractionManager} from 'react-native';
 import Onyx from 'react-native-onyx';
 import _ from 'underscore';
 import * as ReportUtils from '@libs/ReportUtils';
-import CONST from '@src/CONST';
+import Navigation, {navigationRef} from '@navigation/Navigation';
 import ONYXKEYS from '@src/ONYXKEYS';
 import updateUnread from './updateUnread/index';
 
-let previousUnreadCount = 0;
+let allReports = [];
+
+const triggerUnreadUpdate = () => {
+    const currentReportID = navigationRef.isReady() ? Navigation.getTopmostReportId() : '';
+
+    // We want to keep notification count consistent with what can be accessed from the LHN list
+    const unreadReports = _.filter(allReports, (report) => ReportUtils.isUnread(report) && ReportUtils.shouldReportBeInOptionList(report, currentReportID));
+    updateUnread(_.size(unreadReports));
+};
 
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT,
     waitForCollectionCallback: true,
     callback: (reportsFromOnyx) => {
-        if (!reportsFromOnyx) {
-            return;
-        }
-
-        /**
-         * We need to wait until after interactions have finished to update the unread count because otherwise
-         * the unread count will be updated while the interactions/animations are in progress and we don't want
-         * to put more work on the main thread.
-         *
-         * For eg. On web we are manipulating DOM and it makes it a better candidate to wait until after interactions
-         * have finished.
-         *
-         * More info: https://reactnative.dev/docs/interactionmanager
-         */
-        InteractionManager.runAfterInteractions(() => {
-            const unreadReports = _.filter(reportsFromOnyx, (report) => ReportUtils.isUnread(report) && report.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN);
```

Comment:
Why was the `report.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN` condition removed?

Reply:
Because we wanted to sync this exactly with the LHN.

Reply:
I don't know, did we?

I summed it up before as...

 "A notification count should only include threads that are accessible, i.e. can be navigated to".
 
 ...in other words...
 
  "A notification count should **not** include threads that are inaccessible, i.e. **cannot** be navigated to".
 
 ...and I stand with this. It's my personal summary based on my understanding of the issue.
 
Notification preferences are a user-accessible feature:

![image](https://github.com/Expensify/App/assets/2590174/31747bff-e573-47ba-a670-73cf3c020e91)

We don't want to break this, right? It sounds like a different topic.

Reply:
I think this is something different. By setting this preference, you're setting if you want to receive a notification for a new message. Changing this won't really effect the count.

Reply:
In other words, do you suggest that the old code checking for `report.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN` was a mistake (an unnecessary / irrelevant check)?

Or maybe that the user setting I added a screenshot of is not actually related to `report.notificationPreference`?

Reply:
Could be, yes.

Reply:
I investigated this and `CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN` seems to be a special value, and "hidden" reports should not be included in the LHN. It wasn't obvious to me.

We should be fine then

Reply:
I don't know, honestly... Is it possible that this is an incorrectly handled merge conflict?

Reply:
@allroundexperts

Reply:
There are both report and user level notification preferences. But `'hidden'` is indeed a special report level preference that appears when you look at a thread without commenting on it.

Unfortunately, this is a regression. `ReportUtils.shouldReportBeInOptionList()` does not filter out the reports with notification preference `'hidden'` and we **do not want to count those**.

Some suggestions for next time...

- Use git blame and look at _why_ we added something before you assume it is safe to change.
- Still not sure? Ask someone.

Reply:
I concluded wrongly that this check is excessive, as its logic is already present in `shouldReportBeInOptionList`. After re-reading `shouldReportBeInOptionList` it appears clear I was wrong, as it only checks for `isHidden`; a different thing. It might have been a confirmation bias, as Sibtain was quite sure about this change, and he has big experience with the project.
 
> * Use git blame and look at _why_ we added something before you assume it is safe to change.
> 
> * Still not sure? Ask someone.

We made a mistake in being sure enough to push this forward. We're aware of the tools you mention and use them on a daily basis. Also, the compensation system makes sure we're highly motivated to avoid regressions. Still, thank you for your suggestions.

Reply:
@allroundexperts Are we fixing this?

Reply:
@marcaaron Im OoO for two more days. I'll raise a PR for this after then if that's fine.

Reply:
Do we have a GH issue for this mentioned regression?

Reply:
@allroundexperts 

> Im OoO for two more days. I'll raise a PR for this after then if that's fine.

Would you provide an update?

We can move the further discussion to https://github.com/Expensify/App/issues/33506

Reply:
PR created already!